### PR TITLE
fix(api-object): cross-module-version `instanceof` is broken

### LINKED ISSUE #581: Remove `instanceof` use when generating JSON manifest from chart
This is an issue on at least version 2.3.44 and earlier of `cdk8s`. The actual issue experienced was that, when using `aws-cdk-lib/aws-eks.Cluster.addCdk8sChart`, the resulting CDK template had an empty `Manifest` in the custom resource. This occurred after we abstracted out some common low-level CDK8s classes to a library in our monorepo, to make it easier to use the patterns across multiple CDK apps that needed to deploy CDK8s charts to EKS clusters.

I eventually tracked the issue down to the `chartToKube` function's use of `instanceof`: https://github.com/cdk8s-team/cdk8s-core/blob/379953a0928c3cb47392dbabefcde9811de5d571/src/app.ts#L289-L294

When classes in a monorepo have been abstracted out to a lib like this, such that both the lib and the app using it have `cdk8s` as a dependency, `instanceof` fails to correctly identify the class hierarcy as matching when objects are passed between the two. This is the same issue as experienced in `constructs` in https://github.com/aws/constructs/pull/955 As a result, `instanceof` returns false, and the function thinks there are no resources that need jsonifying.

### PR DESCRIPTION
**Pull request by spion**
This pull request fixes the issue with `instanceof` (https://github.com/cdk8s-team/cdk8s-core/issues/581) by implementing `Symbol.hasInstance` (thanks to @fikisipi for pointing it out)

Docs: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol/hasInstance

I welcome any ideas on how to write tests demonstrating the issue is fixed, as to set things up one needs to have multiple versions (or instances) of the same module.

Fixes https://github.com/cdk8s-team/cdk8s-core/issues/581

### CODE REVIEW

File: src/api-object.ts [hunk 1]
```diff
@@ -45,7 +45,29 @@ export interface GroupVersionKind {
...
+  static isApiObject(o: unknown): o is ApiObject {
+    return o != null && typeof o === 'object' && API_OBJECT_SYMBOL in o;
+  }
+
+  /**
+   * Implements `instanceof ApiObject` using the more reliable `ApiObject.isApiObject` static method
+   *
+   * @param o The object to check
+   */
+  static [Symbol.hasInstance](o: unknown) {
```
Comment (iliapolo) [inline src/api-object.ts:69]: This is great! TIL.

Would you mind also implementing this for the [`Chart`](https://github.com/cdk8s-team/cdk8s-core/blob/2.x/src/chart.ts) class? Its also being [used](https://github.com/cdk8s-team/cdk8s-core/blob/2.x/src/app.ts#L124) with `instanceof`, although it does have a `Chart.isChart` function...

File: src/api-object.ts [hunk 2]
```diff
@@ -45,7 +45,29 @@ export interface GroupVersionKind {
...
+   * @param o The object to check
+   */
+  static isApiObject(o: unknown): o is ApiObject {
+    return o != null && typeof o === 'object' && API_OBJECT_SYMBOL in o;
+  }
+
+  /**
+   * Implements `instanceof ApiObject` using the more reliable `ApiObject.isApiObject` static method
+   *
+   * @param o The object to check
```
Comment (iliapolo) [inline src/api-object.ts:66]: Build log suggests you should:

```suggestion
   * @param o The object to check
   * @internal
```

To make jsii happy.

**Review by iliapolo — COMMENTED** [review]
Thanks @spion! 

Yeah I think testing the multi version module scenario is pretty crazy to do in a unit test. I think that we can let it go without it 

👍

File: src/api-object.ts (hunk 2 above)
Comment (spion) [reply src/api-object.ts:66]: Thanks, applied here as well as on `ApiObject`

**Review by iliapolo — APPROVED** [review]
Sweet. Thanks @spion ! 

(notice there are still some build failures)

**Review by iliapolo — APPROVED** [review]

**Pull request merged**